Trying to flesh out Alignment, small commit, build compiles

lckrdg opened the pull request:
Pasha, I'm assuming we don't need any file reads when retrieving index maps anymore, since we store that info in the BinaryGraph. Correct?

I also added the check to not allow duplicate edges in the AdjMatrix, if that's the only duplicates we're rejecting.

Commit:
Trying to flesh out Alignment, small commit, build compiles

Commit:
Actual commit, don't know how the original PR didn't pick these up

khosravipasha commented on src/lib-sana/Alignment/PairwiseAlignment.cpp:
```diff
@@ -0,0 +1,41 @@
+    }
+}
+
+void PairwiseAlignment::printDefinitionErrors(const Graph &G1, const Graph &G2) {
```
would be nice if this could return a `boolean` of whether there was any errors or not.

taylorlanglopez replied:
Ok, will do.

khosravipasha commented on src/lib-sana/Alignment/PairwiseAlignment.cpp:
```diff
@@ -0,0 +1,41 @@
+//Default Constructor and Destructor Already Defined
+PairwiseAlignment::PairwiseAlignment(const PairwiseAlignment &rhs) : A(rhs.A) {}
+
+PairwiseAlignment::PairwiseAlignment(Graph *G1, Graph *G2, const vector<vector<string> > &mapList) {
```
I think similar to what we did for reading graphs, we should read the alignments directly from the file instead of using `vector<vector<string> >`.

khosravipasha replied:
Also moving this function to `Utility` class. And can name the function something like `Utility::LoadPairwiseAlignmentFromEdgeList`.

lckrdg replied:
Ok, I'll work on that next.

khosravipasha reviewed (approved):
Yes, correct we will be storing node names in Graph. (For some bigger graphs we might not want to do this but for most graphs we have right now its ok)

Also put some comments inline for some minor changes. Overall it's good to merge.

Commit:
Made changes suggested by Pasha. Added boolean return type to definition
error check

Commit:
Modified Graph::AddEdge in order to halve the memory cost
Only one way lookup from smaller node to larger node, no duplicates are
allowed.

Commit:
Made an oops by thinking I needed three variables

khosravipasha commented on src/lib-sana/Graph/Graph.cpp:
```diff
@@ -34,10 +34,16 @@ int Graph::RandomNode() {
-    adjLists[node2].push_back(node1);
-    ++numEdges;
+    if (node1 > node2) {
+        unsigned int node1_cpy = node2;  // Const stuff
```
can't we just do `adjLists[node2].push_back(node1);` ?
Not sure why we need a swap, just reversing node1, node2 should work, right?

lckrdg replied:
Absolutely, go ahead and change it. I'm not at home today.

I don't know what I was thinking.

khosravipasha reviewed (approved):
Just added one comment inline. Everything else is great.

Commit:
Changed AddEdge to Pasha's liking

khosravipasha reviewed (approved):
Looks good, thanks 👍 
Will merge it then.
